### PR TITLE
REP-4841 Added an additional GUID for following the Tracing header's value

### PR DESCRIPTION
**Pull request by wdschei**
This insures a truly unique value while still retaining the tracing header's value as part of the UAE's key.

### CODE REVIEW

File: repose-aggregator/components/filters/herp-filter/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala [hunk 1]
```diff
@@ -156,7 +156,7 @@ class HerpFilter @Inject()(configurationService: ConfigurationService,
       "responseCode" -> httpServletResponse.getStatus,
       "responseMessage" -> Try(HttpStatus.valueOf(httpServletResponse.getStatus).name).getOrElse("UNKNOWN"),
       "guid" -> Option(stripHeaderParams(TracingHeaderHelper.getTraceGuid(httpServletRequest.getHeader(CommonHttpHeader.TRACE_GUID.toString))))
-        .getOrElse(java.util.UUID.randomUUID.toString),
+        .getOrElse("").concat(":").concat(java.util.UUID.randomUUID.toString),
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/herp-filter/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala]: I thought we decided to not lead with the colon if we didn't have anything? Maybe we explicitly say we couldn't find the transaction id? So something like `NO_TRANSACTION_ID:{guid}`?

File: repose-aggregator/components/filters/herp-filter/src/test/scala/org/openrepose/filters/herp/HerpFilterTest.scala [hunk 2]
```diff
@@ -138,7 +139,7 @@ class HerpFilterTest extends FunSpec with BeforeAndAfterEach with Matchers with
       // then:
       val logEvents = listAppenderPre.getEvents
       logEvents.size shouldBe 1
-      logEvents.get(0).getMessage.getFormattedMessage should include regex "\"ServiceCode\" : \"\".*\"URL\" : \"\""
+      logEvents.get(0).getMessage.getFormattedMessage should include regex """"ServiceCode" : "".*"URL" : """""
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/herp-filter/src/test/scala/org/openrepose/filters/herp/HerpFilterTest.scala]: SO many quotes. Nothing wrong , it's just funny.

**Comment by adrianjgeorge** [comment]
Of course now it needs a test to prove that's what it's supposed to do.

**Comment by adrianjgeorge** [comment]
Looks good, I think I'll leave the QE task though so we can get another set of eyes on it.

**Comment by wdschei** [comment]
I was still updating the test. It is there now.

**Pull request merged**